cmd/platform_observer/main.go

```diff
@@ -32,19 +32,23 @@ func main() {
 		logger.Fatal("No APIs to observe")
 	}
 
-	minInterval := viper.GetDuration("observer.min_poll")
 	backlogTime := viper.GetDuration("observer.backlog")
+	minInterval := viper.GetDuration("observer.min_poll")
+	maxInterval := viper.GetDuration("observer.max_poll")
+	if minInterval >= maxInterval {
+		logger.Fatal("minimum block polling interval cannot be greater or equal than maximum")
+	}
 
 	var wg sync.WaitGroup
 	wg.Add(len(platform.BlockAPIs))
 
 	for _, api := range platform.BlockAPIs {
 		coin := api.Coin()
 		blockTime := time.Duration(coin.BlockTime) * time.Millisecond
-		pollInterval := blockTime / 4
-
-		if pollInterval < minInterval {
-			pollInterval = minInterval
+		if blockTime < minInterval {
```

Comment:
did you consider moving this into independent function? 

`func (p *Platform)GetBlockPoolInternal() { }` 

easier to test and read.

Please try to not just fix bugs - always improve code

Reply:
Add tests for this

Reply:
`func (p *Platform)GetBlockPoolInternal() { }` per platform is not a good idea, we need to implement that inside all platforms to do the same thing. we can create a function for this, but we need to always get an env var per platform, now we get only one time for all. Another way is to create a function like `func poolInterval(blockTime, min, max time.Duration) time.Duration`

Reply:
👍 do they right thing